react/src/base/inputs/SprkSelect/SprkSelect.js

```diff
@@ -0,0 +1,197 @@
+
+    if (isHugeSelect) {
+      this.setState({
+        selectHugeHasValue: e.target.value !== '',
+      });
+    }
+    if (onChangeFunc) onChangeFunc(e);
+  }
+
+  render() {
+    const {
+      additionalClasses,
+      analyticsString,
+      isDisabled,
+      idString,
+      isValid,
+      id,
+      variant,
+      hasBlankFirstOption,
+      onChange,
+      ariaDescribedBy,
+      choices,
+      ...other
+    } = this.props;
+    const { choiceItems, selectHugeHasValue } = this.state;
+
+    return (
+      <select
+        className={classNames('sprk-b-Select', additionalClasses, {
+          'sprk-b-Input--has-floating-label': selectHugeHasValue,
+          'sprk-b-Select--error': !isValid,
+        })}
+        id={id}
+        disabled={isDisabled}
+        aria-describedby={ariaDescribedBy}
+        onChange={(e) => {
+          this.handleChange(e, variant, onChange);
+        }}
+        data-analytics={analyticsString}
+        data-id={idString}
+        ref={this.selectRef}
+        {...other}
+      >
+        {variant === 'huge' && (
+          // eslint-disable-next-line max-len
+          // eslint-disable-next-line jsx-a11y/control-has-associated-label
+          <option
+            value=""
+            hidden={!hasBlankFirstOption}
+            disabled={!hasBlankFirstOption}
+          />
+        )}
+        {choiceItems.map(
+          ({ id: innerId, label: innerLabel, options, value, ...rest }) => {
+            if (options) {
+              return (
+                <optgroup label={innerLabel} key={innerId}>
+                  {options.map(({ value: optionValue, label: optionLabel }) => (
+                    <option key={optionValue} value={optionValue} {...rest}>
+                      {optionLabel}
+                    </option>
+                  ))}
+                </optgroup>
+              );
+            }
+            return (
+              <option value={value} key={value} {...rest}>
+                {innerLabel}
+              </option>
+            );
+          },
+        )}
+      </select>
+    );
+  }
+}
+
+SprkSelect.propTypes = {
+  /**
+   * A space-separated string of classes to
+   * add to the component.
+   */
+  additionalClasses: PropTypes.string,
+  /**
+   * Assigned to the `data-analytics` attribute serving
+   * as a unique selector for outside libraries
+   * to capture data.
+   */
+  analyticsString: PropTypes.string,
+  /**
+   * Expects an array of objects
+   * that describe the choices available to the user.
+   */
+  choices: PropTypes.arrayOf(
+    PropTypes.shape({
+      /**
+       * Text tied to the rendered input element.
+       */
+      label: PropTypes.string.isRequired,
+      /**
+       * Assigned to the `name` attribute
+       * of the rendered input element.
+       */
+      name: PropTypes.string,
+      /**
+       * Assigned to the `value` attribute
+       * of the rendered input element.
+       */
+      value: PropTypes.string,
```

Comment:
We should add a spread rest here so that they can pass additional attrs through. We had a support ? come up for this.

Reply:
@guern1kn  looks like we still need the spread added here.

Reply:
The component spreads on lines 91, 96 & 104 for these. Did you want me to add a note? We don't document ...rest in the propTypes anywhere.